### PR TITLE
[DX-696] Update field-based-permissions.md

### PR DESCRIPTION
**Pull request by behcio**
### For internal users - please add a Jira DX PR ticket to the subject!
---
### Preview Link

https://deploy-preview-3276--tyk-docs.netlify.app/docs/nightly/graphql/field-based-permissions/

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

### CODE REVIEW

**Review by dcs3spp — COMMENTED** [review]
Hi @behcio Why was the text removed from the PR? Is this targeted for a specific release?

**Comment by behcio** [comment]
> Hi @behcio Why was the text removed from the PR? Is this targeted for a specific release?

This PR was created as part of the training Yaara gave me on the docs. It's a work in progress, so I'll change it to a draft.

**Comment by behcio** [comment]
Please note that these changes in the docs are still not released. Hence the `future-release` label.

File: tyk-docs/content/graphql/field-based-permissions.md [hunk 1]
```diff
@@ -151,12 +151,23 @@ Please note that the list of allowed types overrides the list of restricted type
...
 
 1. *Optional:* Configure a Policy from **System Management > Policies > Add Policy**.
 2. From **System Management > Keys > Add Key** select a policy or configure directly for the key.
 3. Select your GraphQL API (marked as *GraphQL*).
-4. Enable **Field-Based Permissions** for the selected API.
-5. By default all *Types* and *Fields* will be unchecked. By checking a *Type* or *Field* you will disallow to use it for any GraphQL operation associated with the key.
+4. Enable either **Block list** or **Allow list**. It's not possible to have both enabled at the same time.
 
-{{< img src="/img/dashboard/system-management/field_based_permissions.png" alt="field-based-permissions" >}}
+By default all *Types* and *Fields* will be unchecked. By checking a *Type* or *Field* you will disallow to use it for any GraphQL operation associated with the key.
```
Comment (dcs3spp) [inline tyk-docs/content/graphql/field-based-permissions.md:163]:
```suggestion
### Block list

By default all *Types* and *Fields* will be unchecked. By checking a *Type* or *Field* you will disallow to use it for any GraphQL operation associated with the key.
```

File: tyk-docs/content/graphql/field-based-permissions.md [hunk 2]
```diff
@@ -151,12 +151,23 @@ Please note that the list of allowed types overrides the list of restricted type
...
 
-Currently only restricted types and fields can be set up via Tyk Dashboard. Support for allowed types in Tyk Dashboard is coming soon.
+Restricted and allowed types and fields can also be set up via Tyk Dashboard.
 
 1. *Optional:* Configure a Policy from **System Management > Policies > Add Policy**.
 2. From **System Management > Keys > Add Key** select a policy or configure directly for the key.
 3. Select your GraphQL API (marked as *GraphQL*).
-4. Enable **Field-Based Permissions** for the selected API.
-5. By default all *Types* and *Fields* will be unchecked. By checking a *Type* or *Field* you will disallow to use it for any GraphQL operation associated with the key.
+4. Enable either **Block list** or **Allow list**. It's not possible to have both enabled at the same time.
```
Comment (dcs3spp) [inline tyk-docs/content/graphql/field-based-permissions.md]: Is the default behaviour to have *Blocked List* switch enabled with the *Allow List* switch hidden? 
If the *Block List* switch is disabled does this then display and enable the *Allow List* switch?

If so, maybe add an explanation of this in bullet point 4.

File: tyk-docs/content/graphql/field-based-permissions.md [hunk 3]
```diff
@@ -151,12 +151,23 @@ Please note that the list of allowed types overrides the list of restricted type
...
 
-{{< img src="/img/dashboard/system-management/field_based_permissions.png" alt="field-based-permissions" >}}
+By default all *Types* and *Fields* will be unchecked. By checking a *Type* or *Field* you will disallow to use it for any GraphQL operation associated with the key.
+
+For example, the settings from the screenshot below would block `code` and `countries` fields in `Continent` type as well as `latt` and `longt` fields in `Coordinates` type from usage.
```
Comment (dcs3spp) [inline tyk-docs/content/graphql/field-based-permissions.md]:
```suggestion
For example, the settings illustrated below would block the following:
- `code` and `countries` fields in `Continent` type.
- `latt` and `longt` fields in `Coordinates` type.
```

File: tyk-docs/content/graphql/field-based-permissions.md [hunk 4]
```diff
@@ -151,12 +151,23 @@ Please note that the list of allowed types overrides the list of restricted type
...
+
+{{< img src="/img/dashboard/system-management/field-based-permissions-blocklist.png" alt="field-based-permissions" >}}
+
+### Allow list
+
+By default all *Types* and *Fields* will be unchecked. By checking a *Type* or *Field* you will allow to use it for any GraphQL operation associated with the key.
```
Comment (dcs3spp) [inline tyk-docs/content/graphql/field-based-permissions.md]:
```suggestion
By default all *Types* and *Fields* will be unchecked. By checking a *Type* or *Field* you will allow it to be used for any GraphQL operation associated with the key.
```

File: tyk-docs/content/graphql/field-based-permissions.md [hunk 5]
```diff
@@ -151,12 +151,23 @@ Please note that the list of allowed types overrides the list of restricted type
...
+
+For example, the settings from the screenshot below would only allow `code` field in `Continent` type as well as `code` and `name` fields in `Language` to be used. Note that since no query is allowed (no field in `Query` type is checked), you will not be able to run any query whatsoever.
```
Comment (dcs3spp) [inline tyk-docs/content/graphql/field-based-permissions.md]:
```suggestion
For example, the settings illustrated below would only allow the following:
- `code` field in `Continent` type.
- `code` and `name` fields in `Language` type.

Note that the `Query` type is unchecked. Subsequently, you will not be able to run any query.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Hi @behico, PR looks good to me. I have made some minor suggestions. Once the changes have been made and a peer review has been approved I will merge and release

File: tyk-docs/content/graphql/field-based-permissions.md (hunk 2 above)
Comment (behcio) [reply tyk-docs/content/graphql/field-based-permissions.md]: By default neither of the switches is enabled but both are displayed at all times.
However, adding a note that enabling one switch automatically disables the other (if the other was enabled in the first place) as further explanation to `It's not possible to have both enabled at the same time` indeed makes sense.

File: tyk-docs/content/graphql/field-based-permissions.md (hunk 1 above)
Comment (behcio) [reply tyk-docs/content/graphql/field-based-permissions.md:163]: I completely agree with this change but I don't know why `### Block list` was missing in the first place. It's there in commit `6e4da30dcb878755cf60791c443abfb52b1a5714`, line 160. Weird

File: tyk-docs/content/graphql/field-based-permissions.md (hunk 5 above)
Comment (behcio) [reply tyk-docs/content/graphql/field-based-permissions.md]: I like the bullet points here. 
However, I feel some meaning is lost by this change. I wanted to emphasise that having no checked fields in `Query` causes the `Query` type to be unchecked which subsequently makes the user unable to run any query. You could say the first part of this chain is missing.

**Comment by behcio** [comment]
@dcs3spp, I applied changes with your suggestions in mind. Please take a look.

**Review by agata-wit — APPROVED** [review]

**Review by dcs3spp — APPROVED** [review]
Many thanks @behico for submitting changes. I have approved the PR. Is this scheduled for release 5.2.3?

**Comment by behcio** [comment]
> Many thanks @behico for submitting changes. I have approved the PR. Is this scheduled for release 5.2.3?

Rather 5.3.0, but @agata-wit would be the best person to ask.

**Comment by behcio** [comment]
> Many thanks @behico for submitting changes. I have approved the PR. Is this scheduled for release 5.2.3?

Release 5.3.0

**Comment by letzya** [comment]
@dcs3spp why don't we merge this to master?

**Pull request merged**